mlflow/telemetry/events.py

```diff
@@ -230,3 +230,40 @@ def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
 
         model_provider, _ = _parse_model_uri(model_uri)
         return {"model_provider": model_provider}
+
+
+class MakeJudgeEvent(Event):
+    name: str = "make_judge"
+
+    @classmethod
+    def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
+        model = arguments.get("model")
+        if model and isinstance(model, str):
+            model_provider = model.split(":")[0] if ":" in model else None
+            return {"model_provider": model_provider}
+        return {"model_provider": None}
+
+
+class AlignJudgeEvent(Event):
+    name: str = "align_judge"
+
+    @classmethod
+    def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
+        traces = arguments.get("traces")
+        optimizer = arguments.get("optimizer")
+
+        result = {}
+
+        if traces is not None:
+            try:
+                result["trace_count"] = len(traces)
+            except TypeError:
+                result["trace_count"] = None
+
+        if optimizer:
+            optimizer_type = type(optimizer).__name__ if optimizer else None
```

Comment:
```suggestion
            optimizer_type = type(optimizer).__name__
```

`if optimizer` seems redundant.